api/lib/domain/services/user-reconciliation-service.js

```diff
@@ -38,12 +53,36 @@ async function findMatchingSchoolingRegistrationIdForGivenOrganizationIdAndUser(
     throw new NotFoundError('There were no schoolingRegistrations matching with names');
   }
 
-  const matchingSchoolingRegistration = _.find(schoolingRegistrations, { 'id': schoolingRegistrationId });
-  if (!_.isNil(matchingSchoolingRegistration.userId)) {
-    throw new SchoolingRegistrationAlreadyLinkedToUserError();
+  const matchedSchoolingRegistration = _.find(schoolingRegistrations, { 'id': schoolingRegistrationId });
+  await checkIfStudentIsAlreadyReconciledOnTheSameOrganization(matchedSchoolingRegistration, userRepository);
+
+  return matchedSchoolingRegistration;
+}
+
+async function checkIfStudentIsAlreadyReconciledOnTheSameOrganization(matchingSchoolingRegistration, userRepository) {
+  if (!_.isNil(matchingSchoolingRegistration.userId))  {
```

Comment:
C'est pas suffisant de faire juste ça  ?
```suggestion
  if (matchingSchoolingRegistration.userId)  {
```

Reply:
Vu ensemble, on le garde car il teste la nullité de matchingSchoolingRegistration qui peut être undefined